### PR TITLE
fix canvas drawLine bugs

### PR DESCRIPTION
**Pull request by alanwutang11**
There were several bugs with `canvas.drawLine` when using DOM rendering ([this codepath](https://github.com/flutter/engine/blob/main/lib/web_ui/lib/src/engine/html/bitmap_canvas.dart#L412-L416)):

- did not correctly render with negative `ui.Offsets` passed as parameters
- did not respect `paint.strokeCap`
- did not respect `paint.strokeWidth` for horizontal and vertical lines

according to the [api docs for `canvas.drawLine`](https://api.flutter.dev/flutter/dart-ui/Canvas/drawLine.html), rendering of lines should be the same regardless of the `paint.style`.

 Previously, however, failing to set `paint.style` to type `stroke` (either by omission or setting the `paint.style` to `fill`) would cause us to use DOM rendering as specified by [this bool](https://github.com/flutter/engine/blob/a1dd3354058017d73f68af0920b2e8bf66ec818e/lib/web_ui/lib/src/engine/html/bitmap_canvas.dart#L368), which introduced the above mentioned bugs. 

### Solution for negative `ui.Offsets`
instead of using an svg `width`, `height`, and `viewBox` as the bounding box for our path, just remove all of those properties and instead set the svg `overflow` property to `visible`. 

fixes: https://github.com/flutter/flutter/issues/110070

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by mdebbar — COMMENTED** [review]
LGTM but delegating the final stamp of approval to @yjbanov (our SVG expert 🙂)

File: lib/web_ui/test/html/drawing/canvas_lines_golden_test.dart [hunk 1]
```diff
@@ -17,13 +17,17 @@ Future<void> testMain() async {
   const Rect region = Rect.fromLTWH(0, 0, 300, 300);
 
   late BitmapCanvas canvas;
+  late BitmapCanvas domCanvas;
 
   setUp(() {
     canvas = BitmapCanvas(region, RenderStrategy());
+    // seting isInsideSvgFilterTree true forces use of DOM canvas
```
Comment (yjbanov) [inline lib/web_ui/test/html/drawing/canvas_lines_golden_test.dart:24]: nit: setting

**Review by yjbanov — APPROVED** [review]
lgtm

File: lib/web_ui/test/html/drawing/canvas_lines_golden_test.dart (hunk 1 above)
Comment (alanwutang11) [reply lib/web_ui/test/html/drawing/canvas_lines_golden_test.dart]: done

**Pull request merged**